source/simulator/assembly.cc

```diff
@@ -804,6 +913,14 @@ namespace aspect
                                                parameters.composition_degree)
                                               +
                                               (parameters.stokes_velocity_degree+1)/2),
+                                  //TODO: give this a 1-point QGauss to save space, as this is never going to be used here? (discontinuous temperature field doesn't require stabilization)
+                                  QGauss<dim-1>((advection_field.is_temperature()
```

Comment:
But I also have another question: you only treat the case of the temperature equation here, but below you also assemble things for the composition equation. What did we decide: that the DG method is to be used only for the temperature, or for all advected quantities? In either case, either the code here or the assembly below seems wrong.